### PR TITLE
Font Library: fix modal width on mobile viewport

### PR DESCRIPTION
**Pull request by t-hamano**
> **Warning**
> Update: #54471 has been submitted to limit the content width of the modal component. ~~After that PR is merged, I plan to update this PR with that option as well. Until then, please DO NOT MERGE this PR.~~ Merge this PR as a quick fix and refactor this after #54471 is merged. See [this discussion](https://github.com/WordPress/gutenberg/pull/54518#pullrequestreview-1640175993).

## What?

| Before| After |
|--------|--------|
| before | after | 

## Why?

I think the modal should always be displayed full width in mobile viewports.

## How?
I applied the width style via the media query when the viewport width is medium or higher.

### CODE REVIEW

File: packages/edit-site/src/components/global-styles/font-library-modal/index.js
Comment (t-hamano) [inline packages/edit-site/src/components/global-styles/font-library-modal/index.js:46]: Minor improvement

File: packages/edit-site/src/components/global-styles/font-library-modal/style.scss
Comment (jasmussen) [inline packages/edit-site/src/components/global-styles/font-library-modal/style.scss:8]: Can we build this into the modal component itself? Mainly, I'd love to avoid as much local software as possible.

**Review by jasmussen — APPROVED** [review]
Nice! I left a small question, but this is a good catch, thank you!

File: packages/edit-site/src/components/global-styles/font-library-modal/style.scss
Comment (t-hamano) [reply packages/edit-site/src/components/global-styles/font-library-modal/style.scss:8]: Thanks for the review! 

> Can we build this into the modal component itself?

Does this mean that the width is controlled by JavaScript rather than a stylesheet? If so, I can change the approach to something like this, for example:

```javascript
function FontLibraryModal() {
	const isLargeViewport = useViewportMatch( 'medium' );

	return (
		<Modal
			style={ { width: isLargeViewport ? '65vw' : undefined } }
		></Modal>
	);
}
```

Or do you mean adding width constraints to the base modal component itself?

```scss
.components-modal__frame {
	&.is-full-screen {
		@include break-medium() {
			width: 65vw;
		}
	}
}
```

Comment (jasmussen) [reply packages/edit-site/src/components/global-styles/font-library-modal/style.scss:8]: I meant the latter, that ideally we have an excellent _default_ behavior for the modal component, and we provide overrides on a per-component basis only as an exception. 

I've heard someone mention an idea to build in "small", "medium", and "large" props into the modal, so that there's some variety in responsive behaviors. But all that boils down to is to have fewer local overrides, ideally. What do you think?

Comment (t-hamano) [reply packages/edit-site/src/components/global-styles/font-library-modal/style.scss:8]: Well, that might make sense. Currently, there are only two width variations for the modal component: the default auto-width or the full-width with the `isFullScreen` prop opted in.

However, the Font Manager has only recently been implemented, so as the UI and UX improve, the width may change, or this override style itself may become unnecessary, or may be displayed in full screen.

Considering this, I think there is no problem in fixing the bugs and pushing forward with PR for now.

Comment (ciampo) [reply packages/edit-site/src/components/global-styles/font-library-modal/style.scss:8]: Hey all, we've been starting work on this very feature last week — a [PR](https://github.com/WordPress/gutenberg/pull/54471) is up, although I haven't had the time to take a look at it just yet.

Comment (ciampo) [reply packages/edit-site/src/components/global-styles/font-library-modal/style.scss:8]: @t-hamano , I wonder if you'd like to hold on with the work on this PR, in case https://github.com/WordPress/gutenberg/pull/54471 lands and we could use that instead.

Comment (t-hamano) [reply packages/edit-site/src/components/global-styles/font-library-modal/style.scss:8]: @ciampo Thank you for teaching me that. I think #54471 is exactly what this PR needs, so I'm happy to hold this PR .

**Review by ciampo — DISMISSED** [review]
Just temporarily blocking the merge of this PR based on https://github.com/WordPress/gutenberg/pull/54518#discussion_r1328627896

**Comment by matiasbenedetto** [comment]
This PR is looking good, thanks for working on it.


@ciampo is there a estimated time for https://github.com/WordPress/gutenberg/pull/54471 to be shipped?
Otherwise I suggest to merge this, and add a follow-up issue to refactor after https://github.com/WordPress/gutenberg/pull/54471 is ready.

**Comment by ciampo** [comment]
@matiasbenedetto what's the deadline for #54471 to be merged?

**Comment by matiasbenedetto** [comment]
> @matiasbenedetto what's the deadline for https://github.com/WordPress/gutenberg/pull/54471 to be merged?

Font Library is going live in Gutenberg 16.7 so next Monday is the deadline I guess.

File: packages/edit-site/src/components/global-styles/font-library-modal/style.scss
Comment (ciampo) [inline packages/edit-site/src/components/global-styles/font-library-modal/style.scss]: Is there a way we can avoid using `is-full-screen` ? It is an internal classname used by the component and therefore it shouldn't be used by its consumers.

**Review by ciampo — COMMENTED** [review]
In that case, let's go with the plan proposed by @matiasbenedetto — let's merge this PR as a quick fix and refactor it once #54471 gets merged (cc @chad1008 for visibility)

File: packages/edit-site/src/components/global-styles/font-library-modal/style.scss
Comment (t-hamano) [reply packages/edit-site/src/components/global-styles/font-library-modal/style.scss]: This selector is required for increased CSS specificity. However, as I mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/54518#discussion_r1328424969), we can remove this CSS by using the `useViewportMatch` hook instead. What do you think about the hook approach?

Or, I think it can be solved by duplicating the component selectors:

```css
.font-library-modal {
	&.font-library-modal {
		@include break-medium() {
			width: 65vw;
		}
	}
}
```

Comment (ciampo) [reply packages/edit-site/src/components/global-styles/font-library-modal/style.scss]: I'd be ok with any of the two approaches

Comment (t-hamano) [reply packages/edit-site/src/components/global-styles/font-library-modal/style.scss]: OK, I used the approach of nesting `.font-library-modal` selectors. At the same time, I added the TODO comment to remind us to refactor in the future.

**Review by ciampo — COMMENTED** [review]
Thank you! I guess you can go ahead and merge this PR, given that folks were already OK with the styles applied.

**Comment by t-hamano** [comment]
Thank you! I would like to merge this PR since I'm getting the expected results with the updated selector (`&.is-full-screen` > `&.font-library-modal`).

**Comment by mikachan** [comment]
I just cherry-picked this PR to the release/16.7 branch to get it included in the next release: 065bccb9d8

**Pull request merged**
